modules/admin-ui-frontend/test/karma.conf.js

```diff
@@ -78,6 +78,9 @@ module.exports = function (config) {
                     console.error("Suggest installing Firefox or other FOSS browser");
                     throw "No browsers detected";
 	        }
+        if (availableBrowsers.includes('Safari')){
```

Comment:
Indentation seems not to be correct here

Reply:
Why even check if it's there?  Just blindly filter for it.  Filter results would be the same regardless.

Reply:
@gregorydlogan If you build in a non macOS system, Will it create an error if the filter doesn't find anything?

Reply:
The tests will fail if `result` is empty when returned.

Regarding `filter`, I would assume[1] that it returns every item in the list *except* the one you filter for.  Therefore, filtering for 'Safari' should be safe, regardless of whether that term is present in `availableBrowsers`, and the check if it's present prior to filtering is unnecessary.

1: This *is* javascript we're talking about, who the heck knows if `filter` is implemented sanely.

Reply:
@gregorydlogan's comment absolutely makes sense. There is no need for the condition. More than that, we shouldn't modify the list of browsers after we check that there is a valid browser to use. We need to move the modification before the check.

Finally, now that we discuss which browsers are good in basically forever, shouldn't we just stick to known good browsers? So, what about using:

```js
postDetection: (browsers) => {
    const allowed = new Set(['Chrome', 'Firefox', 'ChromeHeadless', 'FirefoxHeadless'])
    console.info('Detected browsers: ' + browsers);
    browsers = browsers.filter(browser => allowed.has(browser));
    console.info('Usable browsers: ' + browsers);
    if (browsers.length < 1) {
        console.error("No browsers detected");
        console.error("Suggest installing Firefox or other FOSS browser");
        throw "No browsers detected";
    }
    return browsers;
}
```